tls/s2n_connection.h

```diff
@@ -49,6 +49,11 @@
 
 #define is_handshake_complete(conn) (APPLICATION_DATA == s2n_conn_get_current_message_type(conn))
 
+typedef enum {
+    S2N_UNBUFFERED_SEND,
```

Comment:
Naming is hard, but this doesn't really seem accurate. We do technically buffer all writes, even in S2N_UNBUFFERED_SEND mode. It's more about how many records we put in the buffer before flushing. Maybe "S2N_IMMEDIATE_SEND"? "S2N_SINGLE_SEND"? "S2N_DEFAULT_SEND"?